provider/equinix/environ.go

```diff
@@ -78,31 +81,100 @@ func (e *environ) Config() *config.Config {
 }
 
 func (e *environ) ConstraintsValidator(ctx context.ProviderCallContext) (constraints.Validator, error) {
-	return nil, errors.NewNotImplemented(nil, "not implemented")
+	validator := constraints.NewValidator()
+	validator.RegisterUnsupported([]string{constraints.CpuPower, constraints.VirtType})
+	validator.RegisterConflicts([]string{constraints.InstanceType}, []string{constraints.Mem})
+	validator.RegisterVocabulary(constraints.Arch, []string{arch.AMD64, arch.ARM64, arch.I386, arch.PPC64EL})
+	return validator, nil
 }
 
 func (e *environ) ControllerInstances(ctx context.ProviderCallContext, controllerUUID string) ([]instance.Id, error) {
-	return nil, errors.NewNotImplemented(nil, "not implemented")
+	insts, err := e.getPacketInstancesByTag(map[string]string{"juju-is-controller": "true", "juju-controller-uuid": controllerUUID})
```

Comment:
You can break this into multiple lines to make it easier to parse the map keys, e.g:

```go
insts, err := e.getPacketInstancesByTag(map[string]string{
  "juju-is-controller": "true", 
   "juju-controller-uuid": controllerUUID,
})
```